### ISSUE TITLE
hanging when saving snapshot after ~epoch 8

### ISSUE THREAD

**Issue by aaronBioBot** [fizyr/keras-retinanet#1354]
Hello, 
I am working through this [tutorial](https://github.com/jaspereb/Retinanet-Tutorial) on using retinanet to train on custom image set on a Google cloud VM, 52 GB RAM, Tesla T4, running Ubuntu 18.04, TF 1.15.2, python 3.7.7 (please let me know if any other information would be helpful). My dataset is comprised of 441 images with 1-5 occrruences of the same object (seeds). The train/val split is 271/170.
I have appeared to correctly install everything. I run the following train command:

`$ nohup keras_retinanet/bin/train.py --tensorboard-dir ~/retinanet/TrainingOutput --snapshot-path ~/retinanet/TrainingOutput/snapshots --random-transform --steps 2000 pascal ~/retinanet/seed_detection_VOC/ & tail -f nohup.out`

Everything works seems to work great for the first few epochs. I can see the loss decrease with tensorboard, etc. But then around epoch 7 or 8 it hangs at this point:

`Epoch 00008: saving model to /home/robotmessenger810/retinanet/TrainingOutput7/snapshots/resnet50_pascal_08.h5`

I have tried increasing RAM on VM (from 30 to 52 GB), setting "--worker 0", and going up and down on the steps (from 100 to 10000). Still the problem persists. Can anyone give insight into what might be happening and how I can troubleshoot?
Thanks!
